src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -6074,6 +6074,13 @@ private bool IsUsingAliasInScope(string name)
                             }
                             else if ((object)leftType != null)
                             {
+                                // Ensure that an error is thrown when an unbound generic is found.
+                                if (right is GenericNameSyntax genericNameRight && rightHasTypeArguments &&
+                                    typeArgumentsSyntax.Any(SyntaxKind.OmittedTypeArgument) && !IsUnboundTypeAllowed(genericNameRight))
```

Comment:
>IsUnboundTypeAllowed [](start = 96, length = 20)

I am confused. It looks like we are binding a generic method invocation, why ```IsUnboundTypeAllowed``` would be relevant here? Also, it is not clear why we check for the error only here, but not, for example, in ```case BoundKind.TypeExpression``` as well. Would it be better to check ```typeArguments``` when we try to bind the invocation?

Reply:
Or perhaps we can check for empty type arguments when we create a BoundMethodGroup node.

---
In reply to: [441136088](https://github.com/dotnet/roslyn/pull/45033#discussion_r441136088) [](ancestors = 441136088)